crates/matrix-sdk-ui/src/timeline/event_item/mod.rs

```diff
@@ -65,6 +65,35 @@ pub(super) enum EventTimelineItemKind {
     Remote(RemoteEventTimelineItem),
 }
 
+/// A wrapper that can contain either a transaction id,
+/// an event id (or both) in order to identify a specific event.
```

Comment:
Sounds like an enum job perhaps

Reply:
Yeah, if it's impossible for both to be `None` we should change this type to an enum (`TransactionId(OwnedTransactionId), EventId(OwnedEventId), Both { ... }`).